src/hooks/polkadotJs/signAndSend.tsx

```diff
@@ -0,0 +1,76 @@
+import { ApolloCache } from '@apollo/client';
+import { ApiPromise } from '@polkadot/api';
+import { SubmittableExtrinsic } from '@polkadot/api/types';
+import { web3FromAddress } from '@polkadot/extension-dapp';
+import { DispatchError, EventRecord } from '@polkadot/types/interfaces';
+import {
+  Callback,
+  ISubmittableResult,
+  RegistryError,
+} from '@polkadot/types/types';
+import { readActiveAccount } from '../accounts/readActiveAccount';
+
+export type ExtrinsicErrors = RegistryError | DispatchError;
+
+export const parseExtrinsicErrors = (
+  events: EventRecord[],
+  apiInstance: ApiPromise
+): ExtrinsicErrors[] =>
+  events
+    .filter(({ event }) => apiInstance.events.system.ExtrinsicFailed.is(event))
+    // we know that data for system.ExtrinsicFailed is
+    // (DispatchError, DispatchInfo)
+    .reduce((acc: ExtrinsicErrors[], { event: { data } }) => {
+      const error: DispatchError = data[0] as DispatchError;
+      if (error.isModule) {
+        // for module errors, we have the section indexed, lookup
+        const decoded = apiInstance.registry.findMetaError(error.asModule);
+        acc.push(decoded);
+      } else {
+        // Other, CannotLookup, BadOrigin, no extra info
+        acc.push(error);
+      }
+
+      return acc;
+    }, []);
+
+export const signAndSend = async (
+  cache: ApolloCache<object>,
+  transaction: SubmittableExtrinsic<'promise', ISubmittableResult>,
+  apiInstance: ApiPromise
+) => {
+  const address = readActiveAccount(cache);
```

Comment:
I think that `readActiveAccount` returns `Account` type, so to avoid confusion, naming should be probably something like:
```
const activeAccount = readActiveAccount(cache);
if (!activeAccount)
....

const { signer } = await web3FromAddress(activeAccount.id);

```